exercises/practice/paasio/src/test/java/PaasioTest.java

```diff
@@ -0,0 +1,337 @@
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class PaasioTest {
+
+    private ByteArrayInputStream dataInputStream;
+    private ByteArrayOutputStream dataOutputStream;
+    private static final String MESSAGECONSTANT = "This is additional Content.";
+
+    @BeforeEach
+    public void setUPTest(TestInfo testInfo) {
+
+        this.dataInputStream = new ByteArrayInputStream("This is data ".getBytes());
+        this.dataOutputStream = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void checkReadOperationCount() {
+
+        try (Paasio customFileReader = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+            customFileReader.read();
+            customFileReader.read();
+            customFileReader.read();
+            assertThat(customFileReader.getReadOperationCount()).isEqualTo(3);
+        } catch (IOException ioException) {
+            System.out.println("Exception occured");
+        }
+
+    }
+
+    @Test
+    void checkAmountOfDataReadInBytes() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            fileOperations.read();
+            fileOperations.read();
+            fileOperations.read();
+            fileOperations.read();
+
+            assertThat(fileOperations.getBytesRead()).isEqualTo(4);
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+    }
+
+    @Test
+    void checkAmountOfDataReadInBytesForMultipleOperations() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            fileOperations.read();
+            fileOperations.read();
+            fileOperations.read();
+            fileOperations.read();
+
+            byte[] readByteData = new byte[5];
+            fileOperations.read(readByteData);
+            assertThat(fileOperations.getBytesRead()).isEqualTo(9);
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+    }
+
+    @Test
+    void checkByteValueReadFromTheFile() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            byte[] fileData = new byte[100];
+            int bytesRead = fileOperations.read(fileData);
+            String dataInFile = new String(fileData, 0, bytesRead, StandardCharsets.UTF_8);
+            assertThat(dataInFile).isEqualTo("This is data ");
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+    }
+
+    @Test
+    void checkIfNullPointerExceptionIsThrownWhenNullIsPassedInsteadOfByteArray() {
+
+        assertThatThrownBy(() -> {
+            Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream);
+            fileOperations.read(null);
+        }).isInstanceOf(NullPointerException.class);
```

Comment:
I think this test case isn't really needed - giving `read` a `null` value isn't particularly interesting.